src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/ReflectMethods.java

```diff
@@ -388,19 +391,30 @@ Name methodName(MethodRef method) {
         return names.fromChars(sigCh, 0, sigCh.length);
     }
 
-    private JCMethodDecl opMethodDecl(Name methodName, CoreOp.FuncOp op) {
-        var mt = new MethodType(com.sun.tools.javac.util.List.nil(), crSyms.funcOpType,
-                com.sun.tools.javac.util.List.nil(), syms.methodClass);
-        var mn = names.fromString("op$").append(methodName);
-        var ms = new MethodSymbol(PUBLIC | STATIC | SYNTHETIC, mn, mt, currentClassSym);
-        currentClassSym.members().enter(ms);
-
-        var opFromStr = make.App(make.Ident(crSyms.opParserFromString),
-                com.sun.tools.javac.util.List.of(make.Literal(op.toText())));
-        var ret = make.Return(make.TypeCast(crSyms.funcOpType, opFromStr));
+    private enum CodeModelStorageOption {
+        TEXT,
+        CODE_BUILDER
+    }
 
-        var md = make.MethodDef(ms, make.Block(0, com.sun.tools.javac.util.List.of(ret)));
-        return md;
+    private JCMethodDecl opMethodDecl(Name methodName, CoreOp.FuncOp op, CodeModelStorageOption codeModelStorageOption) {
+        if (CodeModelStorageOption.TEXT.equals(codeModelStorageOption)) {
```

Comment:
Maybe we can a switch on enum? That should allow you to get rid of the `else` part (as switching on enum constants is "exhaustive")